sources/lib/reasoners/arith.ml

```diff
@@ -40,6 +40,21 @@ module Q = Numbers.Q
 let is_mult h = Sy.equal (Sy.Op Sy.Mult) h
 let mod_symb = Sy.name "@mod"
 
+let calc_power (c : Q.t) (d : Q.t) (ty : Ty.t) =
+  if not (Q.is_int d) then raise Exit;
+  let n = match Z.to_machine_int (Q.to_z d) with
+    | Some n -> n
+    | None -> raise Exit
+  in
+  let sz = Z.numbits (Q.num c) + Z.numbits (Q.den c) in
+  if sz <> 0 && Pervasives.abs n > 100_000 / sz then raise Exit;
```

Comment:
I assume these lines are there to prevent overflow/too big numbers in computations ? If so a comment explaining why and how would be good.